hpat/tests/tests_perf/test_perf_series_str.py

```diff
@@ -0,0 +1,383 @@
+
+    return finish_time - start_time
+
+
+def usecase_series_capitalize(input_data):
+    start_time = time.time()
+
+    input_data.str.capitalize()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_lower(input_data):
+    start_time = time.time()
+
+    input_data.str.lower()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_swapcase(input_data):
+    start_time = time.time()
+
+    input_data.str.swapcase()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_title(input_data):
+    start_time = time.time()
+
+    input_data.str.title()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_upper(input_data):
+    start_time = time.time()
+
+    input_data.str.upper()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_lstrip(input_data):
+    start_time = time.time()
+
+    input_data.str.lstrip()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_rstrip(input_data):
+    start_time = time.time()
+
+    input_data.str.rstrip()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def usecase_series_strip(input_data):
+    start_time = time.time()
+
+    input_data.str.strip()
+
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+@contextmanager
+def do_jit(f):
+    """Context manager to jit function"""
+    cfunc = hpat.jit(f)
+    try:
+        yield cfunc
+    finally:
+        del cfunc
+
+
+def calc_time(f, data):
+    """Calculate execution time of specified function."""
+    start_time = time.time()
+    f(data)
+    finish_time = time.time()
+
+    return finish_time - start_time
+
+
+def calc_compile_time(f, data):
+    """Calculate compile time as difference between first 2 runs."""
+    return calc_time(f, data) - calc_time(f, data)
+
+
+def calc_compilation(pyfunc, data):
+    """Calculate compile time several times."""
+    compile_iteration_number = 5
+
+    compile_times = []
+    for _ in range(compile_iteration_number):
+        with do_jit(pyfunc) as cfunc:
+            compile_times.append(calc_compile_time(cfunc, data))
+
+    return compile_times
+
+
+def calc_results(f, test_data):
+    """Calculate time of boxing+unboxing and internal execution"""
+    test_iteration_number = 5
+
+    exec_times = []
+    boxing_times = []
+    for _ in range(test_iteration_number):
+        ext_start = time.time()
+
+        int_result = f(test_data)
+
+        ext_finish = time.time()
+        exec_times.append(int_result)
+        boxing_times.append(max(ext_finish - ext_start - int_result, 0))
+
+    return exec_times, boxing_times
+
+
+class TestSeriesStringMethods(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.test_results = TestResults()
+        cls.test_results.load()
```

Comment:
I think this is a bad idea to load results from anywhere. If you have really strong requirement for this - may be put it under command line option of variable?